SampleOfferWithoutReward.sol

```diff
@@ -131,6 +131,10 @@ contract SampleOfferWithoutReward {
     // Executing this function before the Offer is signed off by the Client
     // makes no sense as this contract has no money.
     function getDailyPayment() noEther {
+        if (msg.sender != contractor && msg.sender != address(client)) {
```

Comment:
I changed my mind about checking if the sender is the contractor. This checks gives as proof that the contractor still has its private key and no money goes to void. And this is nice property.